### ISSUE TITLE
Multiplayer: Typed Arrays in RPC Functions Throw Errors

### ISSUE THREAD

**Issue by Luzzotica** [godotengine/godot#81306]
### Godot version

4.1.1.stable

### System information

Godot v4.1.1.stable - macOS 13.4.0 - Vulkan (Mobile) - dedicated AMD Radeon Pro 5500M - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 Threads)

### Issue description

Here is the RPC function I have created.

```
rpc("call_local")
func show_items(item_names: Array[String], multiplier: int = 1, reroll: bool = false):
    pass
```

This function throws the following error on the clients (not the RPC caller).

```
E 0:00:19:0744   _process_rpc: RPC - 'CanvasLayer(ItemChooserSystem.gd)::show_items': Cannot convert argument 1 from Array to Array
  <C++ Source>   modules/multiplayer/scene_rpc_interface.cpp:294 @ _process_rpc()
```

I would expect it to be able to convert to the proper type, but getting rid of the type on the array resolves the issue.

### Steps to reproduce

1. Create a multiplayer game.
2. Create an RPC function with a typed array
3. Call that RPC function with a valid array
4. See the error on the client

### Minimal reproduction project

[TypedArrayRPCError.zip](https://github.com/godotengine/godot/files/12515493/TypedArrayRPCError.zip)

**Comment by Jordyfel** [comment]
IIRC this cannot convert array to array error was some other issue, it might not be a problem with rpc here.

**Comment by Jordyfel** [comment]
Related to / duplicate of #74391

**Comment by dalexeev** [comment]
* Duplicate of #69215.
* May be fixed by #78219.

Thanks for reporting the bug nonetheless!
